### ISSUE TITLE
structs as function arguments lacking modifier info

### ISSUE THREAD

**Issue by hedgar2017** [vyperlang/vyper#2609]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.2.dev45+g798ed1a8
* OS: linux
* Python Version (output of `python --version`): 3.10.1

Compiler output:
```
vyper.exceptions.TypeCheckFailure: Statement node did not produce LLL\n\nThis is an unhandled internal compiler error. Please create an issue on Github to notify the developers.\nhttps://github.com/vyperlang/vyper/issues/new?template=bug.md\n[102527] Failed to execute script 'vyper_compile' due to unhandled exception!\n
```

[The test file is here](https://github.com/matter-labs/compiler-tests/blob/main/vyper/simple/algorithm/floating_point_simulation/geometry/area.vy)

**Comment by charles-cooper** [comment]
the issue an unhandled error when we try to assign to function arguments (which are supposed to be immutable)

**Comment by charles-cooper** [comment]
related: https://github.com/vyperlang/vyper/issues/2145#issuecomment-965684394

**Comment by charles-cooper** [comment]
smaller repro:
```python
struct Foo:
    x: uint256

@external
def foo(f: Foo):
    f.x *= 2
```

**Comment by charles-cooper** [comment]
for internal functions, code actually gets generated, it's not just a false positive from the type checker. ex the following code:
```
struct Foo:
    x: uint256
@internal
def foo(f: Foo):
    f.x += 1
```

**Comment by charles-cooper** [comment]
this seems to be fixed as of 046ea166d93b8351da7c5d74115893eea2f60d76
